packages/mui-joy/src/Radio/Radio.tsx

```diff
@@ -204,8 +204,8 @@ const RadioIcon = styled('span', {
   slot: 'Icon',
   overridesResolver: (props, styles) => styles.icon,
 })<{ ownerState: RadioOwnerState }>(({ ownerState }) => ({
-  width: '50%',
-  height: '50%',
+  width: 'calc((var(--Radio-size) - var(--variant-borderWidth)) / 2)',
+  height: 'calc((var(--Radio-size) - var(--variant-borderWidth)) / 2)',
```

Comment:
```suggestion
  width: 'calc((var(--Radio-size) - var(--variant-borderWidth, 0px)) / 2)',
  height: 'calc((var(--Radio-size) - var(--variant-borderWidth, 0px)) / 2)',
```

Fallback value is needed for `--variant-borderWidth` for custom variants.